app/models/manageiq/providers/hawkular/middleware_manager/live_metrics_capture.rb

```diff
@@ -0,0 +1,135 @@
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager::LiveMetricsCapture
+    class TargetValidationError < RuntimeError; end
+    class MetricValidationError < RuntimeError; end
+
+    def initialize(target)
+      @target = target
+      unless @target.kind_of?(MiddlewareServer)
+        raise TargetValidationError, "Validation error: unknown target"
+      end
+      @ems = @target.ext_management_system
+    end
+
+    def metrics_available
+      resource = Struct.new(:id, :feed).new
+      resource.id = @target.nativeid
+      resource.feed = @target.feed
+      available = []
+      @ems.metrics_resource(resource).each do |metric|
+        available << {
+          :id   => metric.id,
+          :name => @target.class::METRICS_HWK_MIQ[metric.name],
+          :type => metric.type,
+          :unit => metric.unit
+        }
+      end
+      available
+    end
+
+    def validate_metric(metric)
+      if metric.nil?
+        false
+      elsif metric[:id].nil?
+        false
+      elsif metric[:name].nil?
+        false
+      elsif metric[:type].nil?
+        false
+      elsif metric[:unit].nil?
+        false
+      else
+        true
+      end
+    end
+
+    def metric(metric, start_time, end_time, interval)
+      unless validate_metric(metric)
+        raise MetricValidationError, "Validation error: metric must be defined"
```

Comment:
If it's always going to raise the exception when called, it can probably go right into the validate_metric method directly.